src/common/IsolatedWorld.ts

```diff
@@ -722,9 +724,7 @@ export class IsolatedWorld {
       waitForVisible: boolean,
       waitForHidden: boolean
     ): Promise<Node | null | boolean> {
-      const node = predicateQueryHandler
-        ? ((await predicateQueryHandler(root, selector)) as Element)
-        : root.querySelector(selector);
+      const node = (await predicateQueryHandler(root, selector)) as Element;
```

Comment:
I think predicateQueryHandler might undefined? If we are sure it's always there, let's update times in makePredicateString to not allow empty predicate query handlers